pennylane/devices/preprocess.py

```diff
@@ -163,6 +163,8 @@ def mid_circuit_measurements(
 
```

Comment:
The new default should be `tree-traversal`

Reply:
That's what I thought at first, but since this PR was opened we adding many features to `one-shot` that are not supported yet. So I'd favor feature over efficiency and let `one-shot` be the default for now.

Reply:
@trbromley What should be the default in your opinion?

Reply:
+1 to favour feature completeness over efficiency, we just need to make sure this option is well represented in the docs.

What are the features missing from `tree-traversal`?

Reply:
I listed the following at the bottom of the description:
- `prng_key`
- `jax.jit`
- broadcasting
to which could be added (soon): supported by Catalyst.

Reply:
`prng_key` support added and tested.